### ISSUE TITLE
A beginner runs into some issues creating SQS queues

### ISSUE THREAD

**Issue by andness** [pulumi/pulumi#4371]
I was testing out Pulumi today and after first having a jolly good time creating AWS resources I accidentally stumbled upon the "serverless building blocks" and thought I could simplify things a lot. But my first attempt at setting up a simple SQS triggered Lambda did not go well.

I started a fresh project with `pulumi new`, picked AWS Typescript and hit enter for all the defaults. I then edited `index.ts` to read: 

    import * as aws from "@pulumi/aws"
     
    const queue = new aws.sqs.Queue("tasks", {
      fifoQueue: true
    })
     
    queue.onEvent("onTask", (e) => {
      console.log("Got an event:", e)
    })

But I made a mistake, which Pulumi caught:

`error: Error validating the FIFO queue name: [FIFO queue name should end with ".fifo": tasks-1696a75]`

Great, I always forget that when making FIFO queues. So I update the name to be `tasks.fifo` and re-run `pulumi up`.

`error: aws:sqs/queue:Queue resource 'tasks.fifo' has a problem: only alphanumeric characters and hyphens allowed in "name"`

Oh... Well that's a bit of pickle 🤔 But since I had previously succeeded in creating a FIFO queue in a different play project I compared the two, and the only difference was that in the successful one I was using the `name:` property, so with fingers crossed I updated the `index.ts` file to read:

```
import * as aws from "@pulumi/aws"

const queue = new aws.sqs.Queue("tasks.fifo", {
  name: "tasks.fifo",
  fifoQueue: true
})

queue.onEvent("onTask", (e) => {
  console.log("Got an event:", e)
})
```

And now ... it hangs. After 5 minutes and 24s it outputs the following:

```
Updating (dev):
     Type                                    Name             Status                  Info
     pulumi:pulumi:Stack                     firstpulumi-dev  **failed**              1 error
 +   └─ aws:sqs:Queue                        tasks.fifo       created
 +      └─ aws:sqs:QueueEventSubscription    onTask           created
 +         ├─ aws:iam:Role                   onTask           created
 +         ├─ aws:iam:RolePolicyAttachment   onTask-7cd09230  created
 +         ├─ aws:iam:RolePolicyAttachment   onTask-32be53a2  created
 +         ├─ aws:lambda:Function            onTask           created
 +         ├─ aws:lambda:Permission          onTask           created
 +         └─ aws:lambda:EventSourceMapping  onTask           **creating failed**     1 error

Diagnostics:
  aws:lambda:EventSourceMapping (onTask):
    error: Error creating Lambda event source mapping: InvalidParameterValueException: Queue visibility timeout: 30 seconds is less than Function timeout: 180 seconds
    {
      Message_: "Queue visibility timeout: 30 seconds is less than Function timeout: 180 seconds",
      Type: "User"
    }

  pulumi:pulumi:Stack (firstpulumi-dev):
    error: update failed

Resources:
    + 7 created
    1 unchanged

Duration: 5m24s
```

That's great, tells me exactly what I did wrong, but I'm wondering why it took 5 minutes to tell me that?

Anyway, I add `visibilityTimeoutSeconds: 180` and re-run, and this time it succeeds. Yay!

So to summarize:
1. It seems the queue creation code should check the FIFO flag before validating the name?
2. Why did it hang for 5 minutes before telling me I got the config wrong?
3. Since Pulumi is managing the creation of the function, shouldn't it be able to get the config to match?

This could just be me being a total beginner, but I figure beginners often make some pretty interesting mistakes so I thought I'd submit it here.

Make no mistake, I'm really enjoying your product so far!

**Comment by leezen** [comment]
Thanks for the awesome feedback. I believe that that error message comes from AWS. One of the things we'd like to do better is have a better semantic understanding of what is/isn't allowed and what configurations can/can't go together. The `awsx` package captures some of that, but in general, we could be doing a much better job of that. I'll leave this issue open as an example of the better configuration checking we could do.

**Comment by stack72** [comment]
Hi @andness 

Sorry for the long silence here - just wanted to circle back here and let you know that in Pulumi-aws v3.19.2, we added support for the autonaming of fifo queues

I hope this helps

Paul

**Comment by stack72** [comment]
After speaking to @leezen, we are going to close this out as we believe the work detailed above fixes the user issues

Paul
